internal/agent/runtime/docker.go

```diff
@@ -0,0 +1,191 @@
+package runtime
+
+import (
+	"context"
+	"fmt"
+	"io"
+	"regexp"
+
+	"github.com/docker/docker/api/types"
+	"github.com/docker/docker/api/types/container"
+	"github.com/docker/docker/api/types/mount"
+	"github.com/docker/docker/client"
+	"github.com/go-logr/logr"
+	"github.com/tinkerbell/tink/internal/agent"
+	"github.com/tinkerbell/tink/internal/agent/runtime/internal"
+	"github.com/tinkerbell/tink/internal/agent/workflow"
+	"github.com/tinkerbell/tink/internal/ptr"
+)
+
+var _ agent.ContainerRuntime = &Docker{}
+
+// Docker is a docker runtime that satisfies agent.ContainerRuntime.
+type Docker struct {
+	log    logr.Logger
+	client *client.Client
+}
+
+// Run satisfies agent.ContainerRuntime.
+func (d *Docker) Run(ctx context.Context, a workflow.Action) error {
+	// We need the image to be available before we can create a container.
+	image, err := d.client.ImagePull(ctx, a.Image, types.ImagePullOptions{})
```

Comment:
might think about adding retries here. a transient issue to the registry could cause a whole workflow to fail.

Reply:
I went with [github.com/avast/retry-go](github.com/avast/retry-go) as it seems quite modular and has all the bits you'd expect from a retry library.